python/cudf/cudf/core/buffer.py

```diff
@@ -33,21 +33,20 @@ class Buffer(Serializable):
         object is kept in this Buffer.
     """
 
-    ptr: int
     size: int
-    _owner: Any
+    _ptr: int
+    _owner: object
 
     def __init__(
-        self, data: Any = None, size: Optional[int] = None, owner: Any = None
+        self, data: Any = None, size: int = None, owner: object = None
```

Comment:
It seems that `size` is only used when `data` is an integer (representing a pointer) and is otherwise ignored. It seems like a more natural API would be to accept a tuple `(pointer, size)` rather than exposing a `size` argument that is ignored by a majority of the code paths.

Is there a reason to keep this API for aligning with some other class(es) outside of this package?

(Not necessary to address in this PR but we may want to open an issue.)

Reply:
Also, is it not required to assign "Optional" types for parameters defaulting to `None`? (`None` is not an `int`, so I'm surprised mypy permits this without complaint.)

Reply:
> Is there a reason to keep this API for aligning with some other class(es) outside of this package?

I don't think so, @vyasr do you know?

> Also, is it not required to assign "Optional" types for parameters defaulting to None? (None is not an int, so I'm surprised mypy permits this without complaint.)

Yes and No. This is what we do throughout the RAPIDS code base and it is allowed by [PEP-484](https://peps.python.org/pep-0484/#union-types). However, I see now that PEP-484 has been changed to recommend the use of `Optional` always.

Personally, I much prefer the only way. I think adding `Optional` to the vast majority of optional arguments bloats the code unnecessary.

Reply:
> I don't think so, @vyasr do you know?

Nope, it should be fine. There is a remote possibility that something in `dask_cudf` or `dask-cuda` is relying on the current behavior, but in either case we own those and can push the fixes downstream if we need to. That said, definitely agree with putting that fix in a subsequent PR.

> Personally, I much prefer the old way. I think adding Optional to the vast majority of optional arguments bloats the code unnecessary.

I prefer having a style guide telling everybody what to do ;) I think we are currently very inconsistent in how we document optional parameters, but we may as well stick to what PEP 484 currently recommends since there's no telling when that will become a mypy error.

Reply:
> I prefer having a style guide telling everybody what to do ;) I think we are currently very inconsistent in how we document optional parameters, but we may as well stick to what PEP 484 currently recommends since there's no telling when that will become a mypy error.

Fair enough :) Maybe we should do a type hint overhauling at some point.

Reply:
@madsbk Can you also discuss why you changed `Any` to `object`? The difference is subtle. I see it [discussed on StackOverflow](https://stackoverflow.com/questions/39817081/typing-any-vs-object) but I'd be interested to hear your rationale.

Reply:
Using `object`, we assume that `owner` has the attributes as `object` and nothing else. Thus, mypy will complain if we do something like `len(owner)`. This is exactly what we want since `Buffer` only uses `owner` to keep a reference to the underlying device memory alive. 
On the other hand when using `Any` we assume that `owner` has **any** attribute thus mypy will accept anything including `len(owner)`.

Reply:
Fantastic! Thanks for that explanation, that aligns with my understanding as well.